tests/unit/test_estimator.py

```diff
@@ -596,6 +600,139 @@ def test_prepare_for_training_force_name_generation(strftime, sagemaker_session)
     assert JOB_NAME == fw._current_job_name
 
 
+def test_git_support_with_branch_and_commit_succeed(sagemaker_session):
```

Comment:
You need to mock your unit tests. Your unit tests should not be able to access git.